app/forms/EnterAgentEORIFormProvider.scala

```diff
@@ -21,11 +21,12 @@ import forms.mappings.Mappings
 import models.EORI
 import play.api.data.Form
 
-class EnterAgentEORIFormProvider @Inject() extends Mappings {
```

Comment:
Why don't you move the TrimSpaces  to  stringFormatter(errorKey: String): Formatter[String] . That way it will come for free for all fields. You can create 2 versions of it one with trimspaces and other without it.

Reply:
Ideally we can't apply trim for all strings using stringFormatter as some fields don't require middle spaces to be trimmed like reason-for-overpayment, declarant ref number and bank account name etc., Please advise.

Reply:
Fixed it.